engine/org.eclipse.birt.report.engine/src/org/eclipse/birt/report/engine/css/engine/StyleConstants.java

```diff
@@ -93,146 +92,150 @@ public interface StyleConstants {
 
 	/** style property: vertical align */
 	int STYLE_VERTICAL_ALIGN = 43;
 
 	/** style property: font weight */
-	int STYLE_FONT_WEIGHT = 44;
+	int STYLE_FONT_WEIGHT = 47;
 
 	/** style property: font variant */
-	int STYLE_FONT_VARIANT = 45;
+	int STYLE_FONT_VARIANT = 48;
 
 	/** style property: master page */
-	int STYLE_MASTER_PAGE = 46;
+	int STYLE_MASTER_PAGE = 44;
 
 	/** style property: border left color */
-	int STYLE_BORDER_LEFT_COLOR = 47;
+	int STYLE_BORDER_LEFT_COLOR = 57;
 
 	/** style property: show if blank */
-	int STYLE_SHOW_IF_BLANK = 48;
+	int STYLE_SHOW_IF_BLANK = 46;
 
 	/** style property: border top style */
-	int STYLE_BORDER_TOP_STYLE = 49;
+	int STYLE_BORDER_TOP_STYLE = 45;
 
 	/** style property: border left width */
-	int STYLE_BORDER_LEFT_WIDTH = 50;
+	int STYLE_BORDER_LEFT_WIDTH = 59;
 
 	/** style property: text overline */
-	int STYLE_TEXT_OVERLINE = 51;
+	int STYLE_TEXT_OVERLINE = 49;
 
 	/** style property: widows */
-	int STYLE_WIDOWS = 52;
+	int STYLE_WIDOWS = 53;
 
 	/** style property: font size */
-	int STYLE_FONT_SIZE = 53;
+	int STYLE_FONT_SIZE = 54;
 
 	/** style property: font style */
-	int STYLE_FONT_STYLE = 54;
+	int STYLE_FONT_STYLE = 55;
 
 	/** style property: background position x */
-	int STYLE_BACKGROUND_POSITION_X = 55;
+	int STYLE_BACKGROUND_POSITION_X = 51;
 
 	/** style property: page break inside */
-	int STYLE_PAGE_BREAK_INSIDE = 56;
+	int STYLE_PAGE_BREAK_INSIDE = 52;
 
 	/** style property: white space */
-	int STYLE_WHITE_SPACE = 57;
+	int STYLE_WHITE_SPACE = 61;
 
 	/** style property: border left style */
-	int STYLE_BORDER_LEFT_STYLE = 58;
+	int STYLE_BORDER_LEFT_STYLE = 65;
 
 	/** style property: background image */
-	int STYLE_BACKGROUND_IMAGE = 59;
+	int STYLE_BACKGROUND_IMAGE = 56;
 
 	/** style property: background image type */
-	int STYLE_BACKGROUND_IMAGE_TYPE = 60;
+	int STYLE_BACKGROUND_IMAGE_TYPE = 58;
 
 	/** style property: background position y */
-	int STYLE_BACKGROUND_POSITION_Y = 61;
+	int STYLE_BACKGROUND_POSITION_Y = 62;
 
 	/** style property: text underline */
-	int STYLE_TEXT_UNDERLINE = 62;
+	int STYLE_TEXT_UNDERLINE = 60;
 
 	/** style property: line height */
-	int STYLE_LINE_HEIGHT = 63;
+	int STYLE_LINE_HEIGHT = 64;
 
 	/** style property: color */
-	int STYLE_COLOR = 64;
+	int STYLE_COLOR = 66;
 
 	/** style property: display */
-	int STYLE_DISPLAY = 65;
+	int STYLE_DISPLAY = 63;
 
 	/** style property: page break after */
-	int STYLE_PAGE_BREAK_AFTER = 66;
+	int STYLE_PAGE_BREAK_AFTER = 67;
 
 	/** style property: font family */
-	int STYLE_FONT_FAMILY = 67;
+	int STYLE_FONT_FAMILY = 69;
 
 	/** style property: background attachment */
 	int STYLE_BACKGROUND_ATTACHMENT = 68;
 
 	/** style property: letter spacing */
-	int STYLE_LETTER_SPACING = 69;
+	int STYLE_LETTER_SPACING = 70;
+
+	/** style property: text hyperlink style */
+	int STYLE_TEXT_HYPERLINK_STYLE = 40;
 
 	/** number (count) of style constants */
-	int NUMBER_OF_STYLE = 70;
```

Comment:
I don't understand why you renumber all constants. Are they not saved somewhere?

Reply:
The ROM version number can be changed. I will do it. 

The constants are changed due to the new indexing of PerfectHash. 
The new ordering of the Word-list with the new attribute crescendo the new Index values. 
This was done by the programme according to the README. MD